pkg/k8s/watchers/pod.go

```diff
@@ -851,6 +851,31 @@ func (k *K8sWatcher) updatePodHostData(oldPod, newPod *slim_corev1.Pod, oldPodIP
 	specEqual := oldPod != nil && newPod.Spec.DeepEqual(&oldPod.Spec)
 	hostIPEqual := oldPod != nil && newPod.Status.HostIP != oldPod.Status.HostIP
 
+	ownerReferenceEqual := false
```

Comment:
Is this strictly necessary? I don't think ownerReferences generally change on existing resources. It's possible, but pretty uncommon (it would generally only happen if the user manually changed it). I'm not opposed to having the logic, but curious if there's an actual real necessity for it.